libs/threading_base/include/hpx/threading_base/scheduler_state.hpp

```diff
@@ -0,0 +1,37 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2011 Bryce Lelbach
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+#if !defined(HPX_THREAD_DATA_SCHEDULER_STATE)
+#define HPX_THREAD_DATA_SCHEDULER_STATE
+
+#include <hpx/config.hpp>
+
+namespace hpx {
+    enum state
+    {
+        state_invalid = -1,
+        state_initialized = 0,
+        first_valid_runtime_state = state_initialized,
+        state_pre_startup = 1,
+        state_startup = 2,
+        state_pre_main = 3,
+        state_starting = 4,
+        state_running = 5,
+        state_suspended = 6,
+        state_pre_sleep = 7,
+        state_sleeping = 8,
+        state_pre_shutdown = 9,
+        state_shutdown = 10,
+        state_stopping = 11,
+        state_terminating = 12,
+        state_stopped = 13,
+        last_valid_runtime_state = state_stopped
+    };
+}
```

Comment:
Have you merged the scheduler state and the runtime state?

Reply:
No. They are merged on master as well. I've only moved the file (or the enum definition). I wanted to separate them initially but decided against it because we e.g. use the scheduler states to compute the thread manager state (before writing this comment I thought the runtime state was computed from the scheduler states), and it's nice to have them more or less synchronized. There are some states which are used by the schedulers but not by the runtime (the ones related to suspension). I'm kind of indifferent on whether they should be the same or not.